animation_nodes/nodes/generic/c_utils.pyx

```diff
@@ -0,0 +1,76 @@
+from ... math cimport lerp, mixVec3, mixEul3, mixMat4, mixQuat, mixColor
+from ... data_structures cimport (
+    ColorList,
+    EulerList,
+    DoubleList,
+    Vector3DList,
+    Matrix4x4List,
+    QuaternionList,
+    VirtualColorList,
+    VirtualEulerList,
+    VirtualDoubleList,
+    VirtualVector3DList,
+    VirtualMatrix4x4List,
+    VirtualQuaternionList
+)
+
+def mixDoubleLists(VirtualDoubleList numbersA, VirtualDoubleList numbersB, VirtualDoubleList factors,
+                   long amount):
+    cdef DoubleList results = DoubleList(length = amount)
+    cdef Py_ssize_t i
+
+    for i in range(amount):
+        results.data[i] = lerp(<float>numbersA.get(i), <float>numbersB.get(i), <float>factors.get(i))
```

Comment:
Add a lerp function for doubles instead of casting here.

Reply:
Okay.